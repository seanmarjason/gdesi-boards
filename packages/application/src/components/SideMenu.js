import * as React from 'react';
import { styled } from '@mui/material/styles';
import Avatar from '@mui/material/Avatar';
import Drawer, { drawerClasses } from '@mui/material/Drawer';
import Box from '@mui/material/Box';
import Divider from '@mui/material/Divider';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';
import SelectContent from './SelectContent';
import MenuContent from './MenuContent';
import OptionsMenu from './OptionsMenu';

const drawerWidth = 240;

export default function SideMenu({ user=null, boardId=null, showMenuContent=true}) {
  return (
    <Drawer
      variant="permanent"
      sx={{
        display: { xs: 'none', md: 'block' },
        [`& .${drawerClasses.paper}`]: {
          backgroundColor: 'background.paper',
        },
      }}
    >
      <Box
        sx={{
          display: 'flex',
          mt: 'calc(var(--template-frame-height, 0px) + 4px)',
          p: 1.5,
        }}
      >
        <SelectContent boardId={boardId}/>
      </Box>
      <Divider />
      <Box
        sx={{
          overflow: 'auto',
          height: '100%',
          display: 'flex',
          flexDirection: 'column',
        }}
      >
        { showMenuContent && <MenuContent boardId={boardId} role={user?.role}/> }
      </Box>
      <Stack
        direction="row"
        sx={{
          p: 2,
          gap: 1,
          alignItems: 'center',
          borderTop: '1px solid',
          borderColor: 'divider',
        }}
      >
        <Avatar
          sizes="small"
          alt={user?.name}
          src="/static/images/avatar/7.jpg"
          sx={{ width: 36, height: 36 }}
        />
        <Box sx={{ mr: 'auto' }}>
          <Typography variant="body2" sx={{ fontWeight: 500, lineHeight: '16px' }}>
            {user?.name}
          </Typography>
          <Typography variant="caption" sx={{ color: 'text.secondary' }}>
            {user?.email}
          </Typography>
        </Box>
        <OptionsMenu />
      </Stack>
    </Drawer>
  );
}
