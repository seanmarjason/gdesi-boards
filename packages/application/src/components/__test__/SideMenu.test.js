import { render } from '@testing-library/react'
import SideMenu from '../SideMenu';

it('SideMenu renders correctly', () => {
  const { container } = render(<SideMenu />)
  expect(container).toMatchSnapshot()
})
