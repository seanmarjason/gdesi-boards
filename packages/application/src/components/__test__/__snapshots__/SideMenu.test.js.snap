// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu renders correctly 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-anchorLeft MuiDrawer-docked css-513su4-MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-1lwhjos-MuiPaper-root-MuiDrawer-paper"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiBox-root css-1idzx9m"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiSelect-root css-19uj6ei-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Select company"
            aria-labelledby="company-select company-simple-select"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="company-simple-select"
            role="combobox"
            tabindex="0"
          >
            <div
              class="MuiListItemAvatar-root css-1ulznza-MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-18k2bs-MuiAvatar-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wdc28j-MuiSvgIcon-root"
                  data-testid="DevicesRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 7c0-.55.45-1 1-1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-1.1 0-2 .9-2 2v11h-.5c-.83 0-1.5.67-1.5 1.5S.67 20 1.5 20H14v-3H4zm19 1h-6c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1m-1 9h-4v-7h4z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
            />
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                aria-hidden="true"
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1kivfpz"
      >
        <div
          class="MuiStack-root css-1bgecpf-MuiStack-root"
        >
          <ul
            class="MuiList-root MuiList-padding MuiList-dense css-b7dj7n-MuiList-root"
          />
        </div>
      </div>
      <div
        class="MuiStack-root css-8lx8d7-MuiStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1qxdgl-MuiAvatar-root"
        >
          <img
            class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
            sizes="small"
            src="/static/images/avatar/7.jpg"
          />
        </div>
        <div
          class="MuiBox-root css-1po99kh"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-19u69in-MuiTypography-root"
          />
          <span
            class="MuiTypography-root MuiTypography-caption css-1374baf-MuiTypography-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
